wgpu-types/src/lib.rs

```diff
@@ -4503,58 +4503,42 @@ pub enum PollType<T> {
     ///
     /// On WebGPU, this has no effect. Callbacks are invoked from the
     /// window event loop.
-    WaitForSubmissionIndex(T),
-
-    /// Same as [`Self::WaitForSubmissionIndex`] but with a timeout.
-    WaitForSubmissionIndexWithTimeout {
+    Wait {
         /// Submission index to wait for.
-        submission_index: T,
+        ///
+        /// If not specified, will wait for the most recent submission at the time of the poll.
+        /// (by the time the method returns, more submissions may have taken place)
+        submission_index: Option<T>,
 
         /// Max time to wait for the submission to complete.
         ///
+        /// It not specified, will indefinitely (or until an error is detected).
         /// If waiting for the GPU device takes this long or longer, the poll will return [`PollError::Timeout`].
-        timeout: Duration,
+        timeout: Option<Duration>,
     },
 
-    /// Same as [`Self::WaitForSubmissionIndex`] but waits for the most recent submission.
-    Wait,
-
-    /// Same as [`Self::Wait`], but with a timeout.
-    ///
-    /// If waiting for the GPU device takes this long or longer, the poll will return [`PollError::Timeout`].
-    WaitWithTimeout(Duration),
-
     /// Check the device for a single time without blocking.
     Poll,
 }
 
 impl<T> PollType<T> {
-    /// Construct a [`Self::Wait`] variant
-    #[must_use]
-    pub fn wait() -> Self {
-        // This function seems a little silly, but it is useful to allow
-        // <https://github.com/gfx-rs/wgpu/pull/5012> to be split up, as
-        // it has meaning in that PR.
-        Self::Wait
-    }
-
-    /// Construct a [`Self::WaitForSubmissionIndex`] variant
+    /// Wait indefinitely until for the most recent submission to complete.
+    ///
+    /// This is a convenience function that creates a [`Self::Wait`] variant with
+    /// no timeout (== wait indefinitely) and no submission index.
     #[must_use]
-    pub fn wait_for(submission_index: T) -> Self {
-        // This function seems a little silly, but it is useful to allow
-        // <https://github.com/gfx-rs/wgpu/pull/5012> to be split up, as
-        // it has meaning in that PR.
-        Self::WaitForSubmissionIndex(submission_index)
+    pub fn wait_indefinitely() -> Self {
```

Comment:
thought: Rust's `std` favors associated constants for cases like this (i.e., `{integer}:::MIN,MAX`, and this feels sorta the same. I'm not inclined to block, just curious if this had been considered.